src/repo-service.cpp

```diff
@@ -42,16 +42,19 @@ void RepoService::stop()
     refresh_timer_->stop();
 }
 
-void RepoService::refresh()
+void RepoService::refresh(const Account *account)
 {
     if (in_refresh_) {
         return;
     }
 
-    const std::vector<Account>& accounts = seafApplet->accountManager()->accounts();
-    if (accounts.empty()) {
-        in_refresh_ = false;
-        return;
+    if (!account) {
+        const std::vector<Account>& accounts = seafApplet->accountManager()->accounts();
+        if (accounts.empty()) {
+            in_refresh_ = false;
```

Comment:
Yeah, but there is still a unnecessary copy when calling `currentAccount`. Why not `hasAccount`?